src/main/java/io/strimzi/kafka/bridge/tracing/TracingHandle.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.tracing;
+
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Simple interface to abstract tracing between legacy OpenTracing and new OpenTelemetry.
+ */
+public interface TracingHandle {
+    /**
+     * Tracing env var service name.
+     *
+     * @return tracing env var service name
+     */
+    String envServiceName();
+
+    /**
+     * Extract service name from bridge confing.
+     *
+     * @param config the bridge config
+     * @return bridge's service name
+     */
+    String serviceName(BridgeConfig config);
+
+    /**
+     * Initialize tracing.
+     */
+    void initialize();
+
+    /**
+     * Adapt executor service if needed.
+     * Else return service parameter instance.
+     *
+     * @param service current executor service
+     * @return adapted executor service or service parameter instance
+     */
+    default ExecutorService adapt(ExecutorService service) {
+        return service;
+    }
+
+    /**
+     * Build span builder handle.
+     *
+     * @param <K> key type
+     * @param <V> value type
+     * @param routingContext Vert.x rounting context
+     * @param operationName current operation name
+     * @return span builder handle
+     */
+    <K, V> SpanBuilderHandle<K, V> builder(RoutingContext routingContext, String operationName);
```

Comment:
I don't understand what purpose this builder serves. It's used once in the `HttpSinkBridgeEndpoint` like this:

```java
            SpanBuilderHandle<K, V> builder = tracing.builder(routingContext, HttpOpenApiOperations.POLL.toString());
            SpanHandle<K, V> span = builder.span(routingContext);
```

So why can't the `builder` method of `TracingHandle` just return the `SpanHandle`? Basically push the existence of builders into the `TracingHandle` implementation as much as possible.

Reply:
It used to be used twice ... with re-impl of consume span handling it reduced to one.

Reply:
OK, removed this mid-builder usage all together.